internal/build/ignition.go

```diff
@@ -38,9 +45,12 @@ var ensureSysextUnit string
 //go:embed templates/k8s-resource-installer.service.tpl
 var k8sResourceUnitTpl string
 
+//go:embed templates/k8s-config-installer.service.tpl
+var k8sConfigUnitTpl string
+
 // configureIngition writes the ignition configuration file based on the provided butane configuration
 // and the given kubernetes configuration
-func (b *Builder) configureIgnition(def *image.Definition, buildDir image.BuildDir, k8sScript string) error {
+func (b *Builder) configureIgnition(def *image.Definition, buildDir image.BuildDir, k8sScript, k8sConfScript string) error {
 	if len(def.ButaneConfig) == 0 && k8sScript == "" {
```

Comment:
I guess here we are missing the check for `k8sConfScript == ""` too. This is mostly to drop a message if no ignition configuration is required at all.